keps/sig-auth/4193-tokenrequest-requester-info-enhancements/README.md

```diff
@@ -0,0 +1,616 @@
+* `serviceaccount_authentication_bound_object_verified_total{bound_object_kind="Node"}`
+* `serviceaccount_bound_tokens_issued_pod_with_node_tokens_total`
+* `serviceaccount_bound_tokens_issued_total{bound_object_kind="Node"}`
+* `serviceaccount_bound_tokens_issued_with_identifier_total`
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+New metrics:
+
+* `serviceaccount_authentication_pod_node_ref_verified_total` - new metric that is incremeneted when a token bound to a Pod has its Node reference verified
+* `serviceaccount_authentication_bound_object_verified_total{bound_object_kind="Node"}` - new metric that is incremeneted when a token bound to a Node has its reference verified
+* `serviceaccount_bound_tokens_issued_pod_with_node_tokens_total` - new metric that is incremented when a node ref is embedded into a bound Pod token (aka implicitly added)
+* `serviceaccount_bound_tokens_issued_total{bound_object_kind="Node"}` - new metric that is incremented whenever a bound token is issued that references a Node (explicitly added)
+* `serviceaccount_bound_tokens_issued_with_identifier_total` - new metric that is incremented whenever a token that contains an identifier/JTI is issued
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Additionally, the metrics detailed above provide a clear signal as to whether these features are being used.
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+###### How can someone using this feature know that it is working for their instance?
+
+For the node info part, using the TokenRequest API and inspecting the contents of the issued JWTs for a token bound to a Pod.
+For JTIs, using the TokenRequest API and then inspecting the contents of the issued JWT for any ServiceAccount token.
+
+For the validation/verification, the user can use the SelfSubjectAccessReview API to check whether the token is still valid.
+To do so, they'd need to obtain a token that is bound to a Pod, delete the corresponding Node object that the Pod is scheduled
+on, and observe that the token is no longer valid via the SelfSubjectAccessReview API.
+
+A similar process could be used for tokens bound to Node objects directly.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+N/A
```

Comment:
Since this isa new alpha things, this is OK, but you've identified a number of metrics in the previous section.  Can a cluster operator use any of the new metrics to determine the health of the new feature or help identify when it needs to be rolled back?

Reply:
Good observation.  As you note, not required yet, but will be important for beta.